### PR TITLE
[BPK-2402] Add theme support to button link

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/Button/Classes/BPKButton.m [hunk 1]
```diff
@@ -415,7 +416,11 @@ - (void)updateContentColor {
         highlightedContentColor = [self class].highlightedOutline;
         break;
     case BPKButtonStyleLink:
-        highlightedContentColor = [self.currentContentColor colorWithAlphaComponent:0.2];
+        if (self.primaryContentColor != nil) {
+            highlightedContentColor = [BPKColor blend:self.linkContentColor with:BPKColor.gray900 weight:0.85f];
```
Comment (k0nserv) [inline Backpack/Button/Classes/BPKButton.m]: Can you expand on this logic? Isn't it enough to reduce opacity here?

Comment (georgegillams) [reply Backpack/Button/Classes/BPKButton.m]: We already darken the colour for highlighted state. I think it should do the same when themed...

Comment (georgegillams) [reply Backpack/Button/Classes/BPKButton.m]: Oh no we don't - I blind 😆

Comment (georgegillams) [reply Backpack/Button/Classes/BPKButton.m]: All fixed @k0nserv

**Pull request merged**